### PR TITLE
Gherkin and automation for specifying a name for  service binding connector

### PR DESCRIPTION
**Pull request by sanketpathak**
Epic: https://issues.redhat.com/browse/ODC-5947
Story: https://issues.redhat.com/browse/ODC-6415

Acceptance criteria:

-  User is able to name the Service Binding connector when creating a Service Binding through the drag & drop action in the Topology view.
-  User can access a Create Service Binding CTA to create a Service Binding from all resources which currently support the drag & drop capability.
-  The Create Service Binding modal issued from the Create Service Binding CTA will allow the user to specify the name as well as the bindable object to connect to.
-  The name is used as the name of the underlying ServiceBindingCR that is created.
-  The name of the ServiceBinding directory in the "/bindings" mountpoint of the Pod matches the name of the ServiceBindingCR (this is the responsibility of the Service Binding Operator).
-  When clicking on the Service Binding Connector in Topology, the associated secret (which is part of the exposed bindable data and is projected into the application's container as files/env variables) should be displayed in the Resources tab of the Side panel

**Checks required for approving Epic gherkin scripts PR:**

- [x] Add @epic-number to the scenarios or feature file [e.g: @odc-xxx]
- [x] Add @to-do for automation possible scenarios
- [x] Add @regression or @smoke based on the importance of the scenario
- [x] Update the test scenarios count in [Automation status confluence Report](https://docs.jboss.org/display/ODC/Automation+Status+Report)
- [x] Check for the linter issues by executing `yarn run gherkin-lint` on frontend folder [Skip epic number tags related linter issues]

**Fix**: https://issues.redhat.com/browse/ODC-6423

**Test Setup**:

Check the TAGS in frontend/packagesmanual or @to-do or @un-verified or @broken-test)"",

Command to execute:

Run a cluster locally and execute the command `yarn run dev` and then `npm run test-cypress-topology` simultaneously

Execute the topology-chart-area-visual.feature/topology/integration-tests/cypress.json file be: ""(@pre-condition or @smoke or @regression) and @5947 and not (@manual or @to-do or @un-verified or @broken-test)"",

Command to execute:

Run a cluster locally and execute the command `yarn run dev` and then `npm run test-cypress-topology` simultaneously

Execute the topology-chart-area-visual.feature file

**Browser**
Chrome 90

**Test Execution Screenshot:**

Screenshot from 2021-12-23 18-41-12

### CODE REVIEW

File: frontend/packages/topology/integration-tests/features/topology/topology-chart-area-visual.feature [hunk 1]
```diff
@@ -315,3 +315,47 @@ Feature: Topology chart area
...
+
+        @regression @manual @odc-5947
+        Scenario: Display of External Bindable resources: T-06-TC28
+            Given user has installed Service Binding operator
+              And user is at developer perspective
+              And user is at Topology page chart view
+              And user has created a deployment workload "node-js"
+            #Please refer to test case KM-01-TC01 for creating kafka connection
+              And user has created external bindable resource Kafka Connection "kafka-instance-ex"
+             When user drag and drop the connector to Kafka Connection
```
Comment (debsmita1) [inline frontend/packages/topology/integration-tests/features/topology/topology-chart-area-visual.feature:340]: when the user drags and drops connector on the bindable service to create the service binding the name field in the modal is pre-populated with the default name for the service binding which the user can edit
default name format : `<source-name>-<source-model-in-lowercase>-<target-name>-<target-model-in-lowercase>`

File: frontend/packages/topology/integration-tests/features/topology/topology-chart-area-visual.feature [hunk 2]
```diff
@@ -315,3 +315,47 @@ Feature: Topology chart area
...
+            Given user has installed Service Binding operator
+              And user is at developer perspective
+              And user is at Topology page chart view
+              And user has created a deployment workload "node-js"
+             When user drag and drop the connector to empty area
+              And user selects Operator backed option
+              And user selects Kafka Connection
+              And user clicks on Create
+              And user clicks on Create button on Create Kafka Connection form
+              And user entered the name "node-kc-connection-1" in Create Service Binding modal
```
Comment (debsmita1) [inline frontend/packages/topology/integration-tests/features/topology/topology-chart-area-visual.feature:359]: same as above, the name field is pre-populated with a default name

**Review by debsmita1 — CHANGES_REQUESTED** [review]

**Comment by debsmita1** [comment]
you need to add the gerkin script for the scenario where there are bindable services available

File: frontend/packages/topology/integration-tests/features/topology/topology-chart-area-visual.feature (hunk 1 above)
Comment (sanketpathak) [reply frontend/packages/topology/integration-tests/features/topology/topology-chart-area-visual.feature:339]: Updated

File: frontend/packages/topology/integration-tests/features/topology/topology-chart-area-visual.feature (hunk 2 above)
Comment (sanketpathak) [reply frontend/packages/topology/integration-tests/features/topology/topology-chart-area-visual.feature]: Updated

File: frontend/packages/topology/integration-tests/features/topology/topology-chart-area-visual.feature [hunk 3]
```diff
@@ -315,3 +315,57 @@ Feature: Topology chart area
...
+        Scenario: Display of External Bindable resources: T-06-TC29
+            Given user has installed Service Binding operator
+              And user is at Topology page chart view
+              And user has created a deployment workload "node-js"
+             When user drag and drop the connector to empty area
+              And user selects Operator backed option
+              And user selects Kafka Connection
+              And user clicks on Create
+              And user clicks on Create button on Create Kafka Connection form
+              And user replaced the name name "node-js-d-kafka-instance-ex-akc" with "node-kc-connection-1" in Create Service Binding modal
```
Comment (debsmita1) [inline frontend/packages/topology/integration-tests/features/topology/topology-chart-area-visual.feature:356]:
```suggestion
              And user replaced the name "node-js-d-kafka-instance-ex-akc" with "node-kc-connection-1" in Create Service Binding modal
```

File: frontend/packages/topology/integration-tests/features/topology/topology-chart-area-visual.feature [hunk 4]
```diff
@@ -315,3 +315,57 @@ Feature: Topology chart area
...
+              And user clicks on Create
+             Then user will see the connection between node workload and Kafka Connection
```
Comment (debsmita1) [inline frontend/packages/topology/integration-tests/features/topology/topology-chart-area-visual.feature:358]: can you also add the step to verify the service binding that is created has the name `node-kc-connection-1`

**Comment by debsmita1** [comment]
Can you add a gerkin script for the scenario where there are bindable services available ?

File: frontend/packages/topology/integration-tests/features/topology/topology-chart-area-visual.feature (hunk 3 above)
Comment (sanketpathak) [reply frontend/packages/topology/integration-tests/features/topology/topology-chart-area-visual.feature]: Updated

File: frontend/packages/topology/integration-tests/features/topology/topology-chart-area-visual.feature (hunk 4 above)
Comment (sanketpathak) [reply frontend/packages/topology/integration-tests/features/topology/topology-chart-area-visual.feature:358]: Updated

File: frontend/packages/topology/integration-tests/features/topology/topology-chart-area-visual.feature [hunk 5]
```diff
@@ -315,3 +315,73 @@ Feature: Topology chart area
...
+              And user is at Topology page chart view
+              And user has created a deployment workload "node-js2"
+            #Please refer to test case KM-01-TC01 for creating kafka connection
+              And user has created external bindable resource Kafka Connection "kafka-instance-ex"
+              And user has created operator-backed service of postgresSQL
+              And user has applied '/testdata/bindableresource1.yaml' yaml
+             When user right clicks on workload "node-js2"
+              And user clicks on "Create Service Binding" option from context menu
+              And uer clicks on Bindable service dropdown
+             Then user will see postgres and kafka connection services options
```
Comment (debsmita1) [inline frontend/packages/topology/integration-tests/features/topology/topology-chart-area-visual.feature:344]: Let the user select a service and Create the service binding

File: frontend/packages/topology/integration-tests/features/topology/topology-chart-area-visual.feature [hunk 6]
```diff
@@ -315,3 +315,74 @@ Feature: Topology chart area
...
+              And user has created a deployment workload "node-js"
+             When user drag and drop the connector to empty area
+              And user selects Operator backed option
+              And user selects Kafka Connection
+              And user clicks on Create
+              And user clicks on Create button on Create Kafka Connection form
+              And user replaced the name "node-js-d-kafka-instance-ex-akc" with "node-kc-connection-1" in Create Service Binding modal
+              And user clicks on Create
+             Then user will see the connection between node workload and Kafka Connection
+              And user will see the name "node-s-d-kafka-instance-ex-akc" in connector sidebar
```
Comment (debsmita1) [inline frontend/packages/topology/integration-tests/features/topology/topology-chart-area-visual.feature:375]:
```suggestion
              And user will see the name "node-kc-connection-1" in connector sidebar
```

Comment (sanketpathak) [reply frontend/packages/topology/integration-tests/features/topology/topology-chart-area-visual.feature]: Updated

File: frontend/packages/topology/integration-tests/features/topology/topology-chart-area-visual.feature [hunk 7]
```diff
@@ -315,3 +315,74 @@ Feature: Topology chart area
...
+            Given user has installed Service Binding operator
+              And user is at Topology page chart view
+              And user has created a deployment workload "node-js2"
+            #Please refer to test case KM-01-TC01 for creating kafka connection
+              And user has created external bindable resource Kafka Connection "kafka-instance-ex"
+              And user has created operator-backed service of postgresSQL "example-pg"
+              And user has applied '/testdata/bindableresource1.yaml' yaml
+             When user right clicks on workload "node-js2"
+              And user clicks on "Create Service Binding" option from context menu
+              And uer clicks on Bindable service dropdown
```
[the remark this answers is not in the document]
Comment (sanketpathak) [reply frontend/packages/topology/integration-tests/features/topology/topology-chart-area-visual.feature]: Updated

File: frontend/packages/topology/integration-tests/features/topology/topology-chart-area-visual.feature [hunk 8]
```diff
@@ -315,3 +315,74 @@ Feature: Topology chart area
...
+        @regression @odc-5947
+        Scenario: Create Service Binding option in nodes actions menu: T-06-TC27
+            Given user has installed Service Binding operator
+              And user is at developer perspective
+              And user is at Topology page chart view
+              And user has created a deployment workload "node-js1"
+             When user right clicks on workload "node-js1"
+              And user clicks on "Create Service Binding" option from context menu
+             Then user will see "Create Service Binding" modal
+              And user will see alert "No bindable services available"
```
[the remark this answers is not in the document]
Comment (sanketpathak) [reply frontend/packages/topology/integration-tests/features/topology/topology-chart-area-visual.feature:329]: It's added at the automation side for closing the modal

**Comment by sanketpathak** [comment]
This PR only affects our e2e tests.
/label qe-approved
/label px-approved
/label docs-approved

**Pull request merged**